app/demo-interactor/KNDemoKernel.cu

```diff
@@ -162,6 +163,10 @@ void initialize(const CudaGridParams&  grid,
                 const StateDeviceRef&  states,
                 const InitialPointers& initial)
 {
+    static const KernelParamCalculator calc_kernel_params(
+        initialize_kernel, "initialize", grid.block_size);
```

Comment:
This is confusing.  One a second call the 'grid' argument would be ignored to construct the `KernelParamCalculator`.

Reply:
I agree, but as the comment above indicates, this is temporary (and in this app the grid is never changed after the user input is read), and the block size isn't currently used except to calculate the occupancy in the output diagnostics. So this should be "fixed" in a future commit. I'll add comments to that effect for the offending use cases.